pkg/apis/rsyslog/validation/rsyslog_relp.go

```diff
@@ -86,6 +89,27 @@ func validateLoggingRules(loggingRules []rsyslog.LoggingRule, fldPath *field.Pat
 	allErrs := field.ErrorList{}
 	if len(loggingRules) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath, "at least one logging rule is required"))
+	} else {
+		for index, rule := range loggingRules {
+			if len(rule.ProgramNames) == 0 && rule.Severity == nil && rule.MessageContent == nil {
+				allErrs = append(allErrs, field.Required(fldPath.Index(index), "at least one of .programNames, .messageContent, or .severity is required"))
+			}
+			if rule.MessageContent != nil {
+				if rule.MessageContent.Regex == nil && rule.MessageContent.Exclude == nil {
+					allErrs = append(allErrs, field.Required(fldPath.Index(index).Child("messageContent"), "either .regex or .exclude has to be provided"))
+				}
+				if regex := rule.MessageContent.Regex; regex != nil {
+					if _, err := regexp.CompilePOSIX(*regex); err != nil {
```

Comment:
Let's use `strconv.Quote` to process the regular expression before compiling it:
```suggestion
					quotedRegex := strconv.Quote(*regex)
					if _, err := regexp.CompilePOSIX(quotedRegex); err != nil {
```

P.S. You could probably extract the check in a method so that it can be reused when checking both `.regex` and `.exclude`